hydra/core/override_parser/types.py

```diff
@@ -191,6 +192,24 @@ def match(s: str, globs: List[str]) -> bool:
         return res
 
 
+class Transformer:
+    @staticmethod
+    def identity(x: ParsedElementType) -> ParsedElementType:
+        return x
+
+    @staticmethod
+    def str(x: ParsedElementType) -> str:
+        return Override._get_value_element_as_str(x)
+
+    @staticmethod
+    def encode(x: ParsedElementType) -> ParsedElementType:
+        # use identity transformation for the primitive types
+        # and str transformation for others
+        if any(isinstance(x, _type) for _type in [str, int, float]):
+            return Transformer.identity(x)
+        return Transformer.str(x)
```

Comment:
1. Why not `isinstance(x, (str, int, float))`
2. What about bool?
3. Why not `isinstance(x, (dict,list))` of naming all the unmodified values?
4. Why return identity(x) and not x?

Reply:
1. Wasn't aware of this syntax :)
2. Oops
3. You mean return `x` for all objects other than dicts and lists?
4. Ahh well in case we ever decide to change the logic for the default case, we will have only one place to change the code